lib/spack/spack/cmd/create.py

```diff
@@ -326,12 +355,14 @@ def __call__(self, stage, url):
         # uses. If the regular expression matches a file contained in the
         # archive, the corresponding build system is assumed.
         clues = [
-            (r'/configure$',      'autotools'),
-            (r'/CMakeLists.txt$', 'cmake'),
-            (r'/SConstruct$',     'scons'),
-            (r'/setup.py$',       'python'),
-            (r'/NAMESPACE$',      'r'),
-            (r'/WORKSPACE$',      'bazel')
+            (r'/configure$',         'autotools'),
+            (r'/configure.(in|ac)$', 'autoreconf'),
+            (r'/Makefile.am$',       'autoreconf'),
+            (r'/CMakeLists.txt$',    'cmake'),
+            (r'/SConstruct$',        'scons'),
+            (r'/setup.py$',          'python'),
+            (r'/NAMESPACE$',         'r'),
+            (r'/WORKSPACE$',         'bazel')
```

Comment:
Previously, `spack create` would loop through all of these and select the last one that matches. By adding the `break` statement below, we save time by choosing the first result that matches. We should discuss the best order for these in case multiple matches are found.